File: judge-route-circle.js
let judgeCircle = function(moves) {
    let strList = moves.split('');
    let LR = 0;
    let UD = 0;
    if(strList.length % 2 !== 0) {
        return false
    } else {
        for(let i of strList) {
            if(i === "U") {
                UD += 1;
            } else if(i === "D") {
                UD -= 1;
            } else if(i === "L") {
                LR += 1;
            } else if(i === "R") {
                LR -= 1;
            }
        }
        if(LR === 0 && UD === 0) {
            return true
        } else {
            return false
        }
    }
};

console.log(judgeCircle('UDD'));

File: judge-route-circle.py
class Solution(object):
    def judgeCircle(self, moves):
    	leftRight = 0
    	upDown = 0
        if(len(moves)%2 != 0 ):
        	return False
        else:
        	for i in moves:
        		if(i == 'U'):
        			upDown +=1
        		elif(i == 'D'):
        			upDown -=1
        		elif(i == 'L'):
        			leftRight += 1
        		elif(i == 'R'):
        			leftRight -= 1
        	if(leftRight == 0 and upDown == 0):
        		return True
        	else:
        		return False

def main():
	moves = "UD"
	print Solution().judgeCircle(moves)

if __name__ == '__main__':
    main()

# solution
def judgeCircle(moves):
    return moves.count('L') == moves.count('R') and moves.count('U') == moves.count('D')

print judgeCircle('UD')